test-data/unit/check-parameter-specification.test

```diff
@@ -2086,3 +2095,59 @@ reveal_type(d(b, f1))  # E: Cannot infer type argument 1 of "d" \
                        # N: Revealed type is "def (*Any, **Any)"
 reveal_type(d(b, f2))  # N: Revealed type is "def (builtins.int)"
 [builtins fixtures/paramspec.pyi]
+
+[case testParamSpecGenericWithNamedArg1]
+from typing import Callable, TypeVar
+from typing_extensions import ParamSpec
+
+T_Retval = TypeVar("T_Retval")
+P = ParamSpec("P")
+
+def run(func: Callable[[], T_Retval], *args: object, backend: str = "asyncio") -> T_Retval:
+    ...
+
+class Result: ...
+def run_portal() -> Result: ...
+
+def submit(
+    func: Callable[P, T_Retval],
+    /,
+    *args: P.args,
+    **kwargs: P.kwargs,
+) -> T_Retval: ...
+
+reveal_type(submit(  # N: Revealed type is "__main__.Result"
+    run,
+    run_portal,
+    backend="asyncio",
+))
+submit(
+    run,  # E: Argument 1 to "submit" has incompatible type "Callable[[Callable[[], T_Retval], VarArg(object), DefaultNamedArg(str, 'backend')], T_Retval]"; expected "Callable[[Callable[[], Result], int], Result]"
```

Comment:
Also off-topic for this PR, but I find it unfortunate that we use these `mypy_extensions` types in error messages, given that we [document them as deprecated](https://mypy.readthedocs.io/en/stable/additional_features.html#extended-callable-types) :(

Reply:
I actually was thinking about the same thing myself as I was looking at this test. This is not hard, but tedious. It would be great if someone will do this.